compose-cluster/generate_compose_yaml.py

```diff
@@ -0,0 +1,258 @@
+	f_init.write('echo "Host config0 is up"\n')
+	
+	config_init_doc = {}
+	config_init_doc['_id'] = "ConfigRS"
+	config_init_doc['configsvr'] = True
+	config_init_doc['members'] = []
+	for config_index in range(CONFIGDB_REPLICA_COUNT):
+		config_init_doc['members'].append({'_id' : config_index, 'host' : 'config{}:27017'.format(config_index)})
+	
+	f_init.write("/mongo_rs_initiate.sh config0 '{}'\n".format(json.dumps(config_init_doc)))
+	f_init.write('echo "ConfigDB replicas have been configured"\n')
+	
+	#Configure replica 0 for each shard
+	for shard_index in range(MONGO_SHARD_COUNT):
+		f_init.write("/wait_for_mongo.sh s{}r0\n".format(shard_index))
+		f_init.write('echo "Host s{}r0 is up"\n'.format(shard_index))
+		
+		shard_init_doc = {}
+		shard_init_doc['_id'] = "MongoRS{}".format(shard_index)
+		shard_init_doc['members'] = []
+		for replica_index in range(MONGO_REPLICA_COUNT):
+			shard_init_doc['members'].append({'_id' : replica_index, 'host' : "s{}r{}".format(shard_index, replica_index)})
+		
+		f_init.write("/mongo_rs_initiate.sh s{}r0 '{}'\n".format(shard_index, json.dumps(shard_init_doc)))
+		f_init.write('echo "Shard-{} replicas have been configured"\n'.format(shard_index))
+	
+	f_init.write("/wait_for_mongo.sh router\n")
+	f_init.write('echo "Host router is up"\n')
+	
+	for shard_index in range(MONGO_SHARD_COUNT):
+		shard_config = "MongoRS{}/".format(shard_index)
+		for replica_index in range(MONGO_REPLICA_COUNT):
+			shard_config += "s{}r{}:27017,".format(shard_index, replica_index)
+		shard_config = shard_config.rstrip(',')
+		f_init.write("/mongo_add_shard.sh router '{}'\n".format(shard_config))
+	
+	f_init.write('echo "All shards have been added to the MongoDB cluster..."\n')
+	f_init.write('echo "Done."\n')
+
+
+#Configure the initializer container
+yaml_obj['services']['initializer'] = {}
+yaml_obj['services']['initializer']['build'] = {}
+yaml_obj['services']['initializer']['build']['context'] = "initializer"
+
+yaml_obj['services']['initializer']['networks'] = {}
+yaml_obj['services']['initializer']['networks']['internalnetwork'] = {}
+yaml_obj['services']['initializer']['networks']['internalnetwork']['aliases'] = ['initializer']
+
+yaml_obj['services']['initializer']['depends_on'] = ['router']
+
+#Configure the initial LFS container
+print("Configuring service: lfsinitial")
+yaml_obj['services']['lfsinitial'] = {}
+yaml_obj['services']['lfsinitial']['image'] = "lfs/lfs:{}".format(LFS_DOCKER_TAG)
+
+yaml_obj['services']['lfsinitial']['networks'] = {}
+yaml_obj['services']['lfsinitial']['networks']['internalnetwork'] = {}
+yaml_obj['services']['lfsinitial']['networks']['internalnetwork']['aliases'] = ['lfsinitial']
+
+yaml_obj['services']['lfsinitial']['environment'] = []
+yaml_obj['services']['lfsinitial']['environment'].append("INITIAL_SLING_NODE=true")
+yaml_obj['services']['lfsinitial']['environment'].append("INSIDE_DOCKER_COMPOSE=true")
+yaml_obj['services']['lfsinitial']['environment'].append("LFS_RELOAD=$LFS_RELOAD")
```

Comment:
This seems to fix the warning:

```suggestion
yaml_obj['services']['lfsinitial']['environment'].append("LFS_RELOAD=${LFS_RELOAD:-}")
```

Reply:
This change appears to break it, for me:
```
fnguyen@fnguyen-PS42-8M:~/Documents/lfs/compose-cluster$ docker-compose build
ERROR: Invalid interpolation format for "environment" option in service "lfsinitial": "LFS_RELOAD=${LFS_RELOAD:-}"
```